### PR TITLE
Fixes #23419: Bad doc for known authentication backends

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/23419

### CODE REVIEW

File: auth-backends/README.adoc [hunk 1]
```diff
@@ -447,7 +447,7 @@ Rudder support plain old `OAUTHv2` and `OpentID Connect`. They have several norm
 [notice]
 ====
 
-We advise to use OICD over SAMLv2 if possible.
+We advise to use OICD over OauthV2 if possible.
```
Comment (amousset) [inline auth-backends/README.adoc:450]:
```suggestion
We advise to use OICD over OAuth 2.0 if possible.
```

Comment (amousset) [inline auth-backends/README.adoc:450]:
```suggestion
We advise to use OIDC over OauthV2 if possible.
```

File: auth-backends/README.adoc [hunk 2]
```diff
@@ -12,7 +12,7 @@ other general information.
 
 = Authentication backends
 
-This plugins allows to use alternative authentication backends for Rudder: *SAMLv2*, *OpenID Connect*, and *LDAP or Active Directory (AD)*.
+This plugins allows to use alternative authentication backends for Rudder: *OAuthV2*, *OpenID Connect*, and *LDAP or Active Directory (AD)*.
```
[the remark this answers is not in the document]
Comment (amousset) [reply auth-backends/README.adoc:15]: and a `sed 's/Oauth/OAuth/` on the rest of the file would make me happy

**Comment by fanf** [comment]
PR updated with a new commit

**Review by amousset — APPROVED** [review]

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**